src/core/eval.ml

```diff
@@ -331,27 +343,32 @@ and snf : term -> term = fun t ->
   | Wild        -> assert false
   | TRef(_)     -> assert false
 
+(** [snf ctx t] computes the strong normal form of term [t] in context
+    [ctx]. *)
+let snf : ctxt -> term -> term = fun ctx t -> snf (Ctxt.to_llet ctx t)
+
 (** [whnf t] computes a weak head-normal form of [t]. *)
-let whnf : term -> term = fun t ->
+let whnf : ctxt -> term -> term = fun ctx t ->
   Stdlib.(steps := 0);
   let t = unfold t in
-  let u = whnf t in
+  let u = whnf (Ctxt.to_llet ctx t) in
   if Stdlib.(!steps = 0) then t else u
```

Comment:
it should be `then unfold t`

Reply:
If I remember well, the position of the `unfold` influenced performance more than Rodolphe expected.

Reply:
Your modification is ok here.